index.ts

```diff
@@ -298,6 +304,13 @@ export class OAuth2AuthCodePKCE {
       + `code_challenge=${encodeURIComponent(codeChallenge)}&`
       + `code_challenge_method=S256`;
 
+    if (extraAuthorizeParams) {
+      url = Object.entries(extraAuthorizeParams).reduce(
+          (builtUrl: string, [key, val]: [string, string]) =>
+              builtUrl+`&${key}=${encodeURIComponent(val)}`,
+          url)
+    }
+
```

Comment:
Line 376 has the same process. Please create a function with abstracts this process and to prevent accidental mis-behaviors.

```
url = `${url}&${dictToQueryString(extraAuthorizeParams)}`

function dictToQueryString(dict: {[k: string]: string}): string { ... }
```